pontoon/base/tests/test_fluent.py

```diff
@@ -193,3 +199,145 @@ def test_flat_transformer_value_and_attributes():
 def test_get_simple_preview(k):
     string, expected = SIMPLE_TRANSLATION_TESTS[k]
     assert get_simple_preview(string) == expected
+
+
+def test_is_plural_expression_not_select_expression():
+    # Return False for elements that are not select expressions
+    input = dedent(
+        """
+        my-entry = Hello!
+    """
+    )
+
+    message = parser.parse_entry(input)
+    element = message.value.elements[0]
+
+    assert is_plural_expression(element) is False
+
+
+def test_is_plural_expression_all_cldr_plurals():
+    # Return True if all variant keys are CLDR plurals
+    input = dedent(
+        """
+        my-entry =
+            { $num ->
+                [one] Hello!
+               *[two] World!
+            }`
+    """
+    )
+
+    message = parser.parse_entry(input)
+    element = message.value.elements[0]
+
+    assert is_plural_expression(element.expression) is True
+
+
+def test_is_plural_expression_all_numbers():
+    # Return True if all variant keys are numbers
+    input = dedent(
+        """
+        my-entry =
+            { $num ->
+                [1] Hello!
+               *[2] World!
+            }`
```

Comment:
If we do have selectors like this, should we really generate plurals even though the source doesn't include even `other`? To me, this looks like a selector on a number, but not a plural selector.

Reply:
I'm open to that change, however I'm just following the logic we use in the frontend. If we were to make a change, it needs to happen on both ends.

Reply:
Added #2805 for this.